File: src/main/java/random/telegramhomebot/auth/db/repositories/PrivilegeRepository.java
package random.telegramhomebot.auth.db.repositories;

import org.springframework.data.jpa.repository.JpaRepository;
import random.telegramhomebot.auth.db.entities.Privilege;

public interface PrivilegeRepository extends JpaRepository<Privilege, Long> {

    Privilege findByName(String name);
}

File: src/main/java/random/telegramhomebot/auth/db/repositories/PrivilegeRepository.kt
package random.telegramhomebot.auth.db.repositories

import org.springframework.data.r2dbc.repository.R2dbcRepository
import random.telegramhomebot.auth.db.entities.Privilege
import reactor.core.publisher.Mono

interface PrivilegeRepository : R2dbcRepository<Privilege, Long> {
    fun findByName(name: String): Mono<Privilege>
}
